### PR TITLE
Add basic bundle resigning support

### PR DESCRIPTION
**Pull request by ejoerns**
This adds the command `rauc resign` which allows to replace the bundle signature without modifying the bundle content.

Based on PR #152

### CODE REVIEW

File: src/bundle.c [hunk 1]
```diff
@@ -265,6 +265,90 @@ gboolean create_bundle(const gchar *bundlename, const gchar *contentdir, GError
...
+		goto out;
+	}
+
+	res = g_seekable_truncate(G_SEEKABLE(outstream), size, NULL, &ierror);
+	if (!res) {
+		g_propagate_error(error, ierror);
+		goto out;
+	}
+
+
```
Comment (jluebbe) [inline src/bundle.c]: Remove the duplicate empty line.

File: src/bundle.c [hunk 2]
```diff
@@ -265,6 +265,90 @@ gboolean create_bundle(const gchar *bundlename, const gchar *contentdir, GError
 	return res;
 }
 
+static gboolean truncate_bundle(const gchar *inpath, const gchar *outpath, gsize size, GError **error) {
```
Comment (jluebbe) [inline src/bundle.c]: This just copies the first size bytes from inpath to outpath. Is there no simpler way?

File: test/bundle.c [hunk 3]
```diff
@@ -130,14 +130,49 @@ static void bundle_test_extract_manifest(BundleFixture *fixture,
...
+	g_assert_false(check_bundle(fixture->bundlename, &size, TRUE, NULL));
+
+	r_context()->config->keyring_path = g_strdup("test/openssl-ca/dev-ca.pem");
+	res = resign_bundle(fixture->bundlename, resignbundle, &ierror);
+	g_assert_no_error(ierror);
+	g_assert_true(res);
+
+	/* Verify resigned bundle with dev keyring */
+	r_context()->config->keyring_path = g_strdup("test/openssl-ca/dev-ca.pem");
+	g_assert_true(check_bundle(resignbundle, &size, TRUE, NULL));
```
Comment (jluebbe) [inline test/bundle.c]: Add a comment why this still verifies against the dev keyring.

**Comment by ejoerns** [comment]
@jluebbe I updated the PR considering your comments

**Pull request merged**